### PR TITLE
Add: Typing Sound Audio Bus in Character Settings

### PR DESCRIPTION
**Pull request by anis-mahsoume**
Add a new field "Audio Bus" into the character settings typing sounds section:
image
The Audio Bus is an OptionButton field which values are populated from AudioServer buses.
I decided to make the change after opening the issue:
#2420

### CODE REVIEW

File: addons/dialogic/Modules/Text/character_settings/character_moods_settings.gd [hunk 1]
```diff
@@ -228,3 +230,10 @@ func preview() -> void:
 		await preview_timer.timeout
 
 	preview_timer.queue_free()
+
+
+func _load_audio_bus_options() -> void:
+	%AudioBus.clear()
+	for i in AudioServer.bus_count:
```
Comment (CakeVR) [inline addons/dialogic/Modules/Text/character_settings/character_moods_settings.gd:237]:
```suggestion
	for i: int in AudioServer.bus_count:
```

File: addons/dialogic/Modules/Text/character_settings/character_moods_settings.tscn [hunk 2]
```diff
@@ -163,6 +163,18 @@ size_flags_horizontal = 3
...
 file_mode = 3
 
+[node name="AudioBusLabel" type="Label" parent="VBox/Settings/Grid"]
+layout_mode = 2
+text = "Audio Bus:"
+
+[node name="AudioBus" type="OptionButton" parent="VBox/Settings/Grid"]
+unique_name_in_owner = true
+layout_mode = 2
+tooltip_text = "Audio bus in which the type sounds will be sent"
```
Comment (CakeVR) [inline addons/dialogic/Modules/Text/character_settings/character_moods_settings.tscn:173]:
```suggestion
tooltip_text = "Audio bus on which the typing sounds will be played."
```

File: addons/dialogic/Modules/Text/node_type_sound.gd [hunk 3]
```diff
@@ -41,6 +41,9 @@ func _ready() -> void:
 
 
 func _on_started_revealing_text() -> void:
+	var default_bus_index: int = AudioServer.get_bus_index(bus)
+	var bus_index: int = current_overwrite_data.get('audio_bus', default_bus_index)
+	bus = AudioServer.get_bus_name(bus_index)
 	if !enabled or (get_parent() is DialogicNode_DialogText and !get_parent().enabled):
```
Comment (CakeVR) [inline addons/dialogic/Modules/Text/node_type_sound.gd:47]:
```suggestion
	bus = AudioServer.get_bus_name(bus_index)
	
	if !enabled or (get_parent() is DialogicNode_DialogText and !get_parent().enabled):
```

**Review by CakeVR — CHANGES_REQUESTED** [review]
Just some minor improvements, you can simply apply all of them.

Thanks for your Pull Request.

**Comment by anis-mahsoume** [comment]
> Just some minor improvements, you can simply apply all of them.
> 
> Thanks for your Pull Request.

Thanks @CakeVR for your review: I have applied the changes you suggested.

**Comment by CakeVR** [comment]
The problem I have with this solution is that you use `_on_started_revealing_text` to set the Audio Bus which gets triggered every single text message.

I don't think this is a good solution.

**Comment by Jowan-Spooner** [comment]
I don't think this should be a setting in the character editor, rather a global setting or a style setting. I cannot see a use-case where you'd need to change this per type sound mood.

**Comment by CakeVR** [comment]
Hello, is there any interest in continuing on this PR?
As I agree with Jowan, there would have to be Global Setting in the Settings tab instead.

**Comment by anis-mahsoume** [comment]
Hi,
This issue has been fixed in
https://github.com/dialogic-godot/dialogic/pull/2456
So closing this PR

**Pull request closed without merging**